### PR TITLE
Add scharr gradient operator.

### PR DESCRIPTION
**Pull request by windelbouwman**

### CODE REVIEW

File: src/gradients.rs [hunk 1]
```diff
@@ -50,6 +50,34 @@ pub fn vertical_sobel(image: &GrayImage) -> Image<Luma<i16>> {
     filter3x3(image, &VERTICAL_SOBEL)
 }
 
+/// Scharr filter for detecting horizontal gradients.
+///
+/// Used by the [`horizontal_scharr`](fn.horizontal_scharr.html) function.
+pub static HORIZONTAL_SCHARR: [i32; 9] = [
+    3,  0,  -3,
```
Comment (theotherphil) [inline src/gradients.rs]: Both of these filters are in the opposite direction to the Sobel and Prewitt filters. Also, could you stick all the filters next to each other at the top of the file.

**Comment by theotherphil** [comment]
Thanks! I’ll merge once the orientations have been flipped to match the other filters in this file.

File: examples/scharr.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    let vertical_scharr_gradient = vertical_scharr(&input_image);
+    save_gradient(
+        &vertical_scharr_gradient,
+        &output_dir,
+        "scharr_vertical.png",
+    );
+}
+
+fn save_gradient(gradient: &Image<Luma<i16>>, output_dir: &Path, filename: &str) {
```
Comment (windelbouwman) [inline examples/scharr.rs]: I added this example, but I cannot figure out how to properly pass the `Luma<i16>` typed image to this function. What should I use for this?

**Comment by theotherphil** [comment]
Sorry, I saw this PR when on my phone a few day ago and then forgot to reply when I got home.

I wasn't very clear about "opposite direction" in my initial comment. I didn't mean to swap the definitions of horizontal and vertical, but rather to swap the signs of all the values in the filters. The Sobel and Prewitt filters have the negative values on the left/top, while your new filters have the negative values on the right/bottom.

File: examples/scharr.rs (hunk 2 above)
Comment (theotherphil) [reply examples/scharr.rs]: I'm not clear what you mean here. What error are you getting?

**Comment by theotherphil** [comment]
Ping @windelbouwman. Are you still intending to work on this?

I can fix it up and get it merged if you’ve abandoned the PR.

**Comment by windelbouwman** [comment]
@theotherphil I have abandoned this PR, please feel free to merge / use it!

**Comment by theotherphil** [comment]
Ok, thanks for the update. I'll fix the outstanding issue and merge later this week.

**Comment by theotherphil** [comment]
Superseded by https://github.com/image-rs/imageproc/pull/353

**Pull request closed without merging**